src/PasswordInput/index.jsx

```diff
@@ -7,140 +7,50 @@
-         */
-        forceHover           : PropTypes.bool,
-        /**
-         *  Display as error/invalid
-         */
-        hasError             : PropTypes.bool,
-        /**
-         *  Display Button icon as disabled
-         */
-        iconButtonIsDisabled : PropTypes.bool,
-        /**
-         *  Alignment of the icon
-         */
-        iconPosition         : PropTypes.oneOf( [ 'left', 'right' ] ),
-        /**
-         *  Component id
-         */
-        id                   : PropTypes.string,
-        /**
-         *  Callback that receives the native <input>: ( ref ) => { ... }
-         */
-        inputRef             : PropTypes.func,
-        /**
-         *  Display as disabled
-         */
-        isDisabled           : PropTypes.bool,
-        /**
-         *  Display as read-only
-         */
-        isReadOnly           : PropTypes.bool,
-        /**
-         *  HTML name attribute
-         */
-        name                 : PropTypes.string,
-        /**
-         *  Input change callback function
-         */
-        onChangeInput        : PropTypes.func,
-        /**
-         *  Icon click callback function
-         */
-        onClickIcon          : PropTypes.func,
-        /**
-         *  Show password as plain text
-         */
-        passwordIsVisible    : PropTypes.func,
-        /**
-         *  Placeholder text
-         */
-        placeholder          : PropTypes.string,
-        /**
-         *  Input text alignment
-         */
-        textAlign            : PropTypes.oneOf( [ 'auto', 'left', 'right' ] ),
-        /**
-         *  Input string value
-         */
-        value                : PropTypes.string,
-    };
-
-    static defaultProps =
-    {
-        aria                 : undefined,
-        className            : undefined,
-        cssMap               : undefined,
-        forceHover           : false,
-        hasError             : false,
-        iconButtonIsDisabled : undefined,
-        iconPosition         : 'right',
-        id                   : undefined,
-        inputRef             : undefined,
-        isDisabled           : false,
-        isReadOnly           : false,
-        name                 : undefined,
-        onChangeInput        : undefined,
-        onClickIcon          : undefined,
-        passwordIsVisible    : false,
-        placeholder          : undefined,
-        textAlign            : 'auto',
-        value                : '',
-    };
-
-    constructor( props )
-    {
-        super();
 
-        this.state = {
-            id                : props.id || generateId( 'PasswordInput' ),
-            passwordIsVisible : false,
-        };
+    const [ passwordIsVisibleState,
+        setPasswordIsVisibleState ] = useState( false );
 
-        this.handleClickIcon = this.handleClickIcon.bind( this );
-    }
+    const passwordRef = useRef();
+
+    useImperativeHandle( ref, () => ( {
```

Comment:
We don’t need to implement this here, since it is (or _should be_) already implemented by the TextInputWithIcon component. 

All we need to do in this component is pass `ref` to the TextInputWithIcon, that is:
```
return (
    <TextInputWithIcon
        { ...props }
        ref = { ref }
        .... />
);
 ```